### ISSUE TITLE
File-scoped Namespaces in C#

### ISSUE THREAD

**Issue by Jerrill** [doxygen/doxygen#9317]
C# code that uses file-scoped namespaces as follows are not being correctly nested under their namespace in the left-nav menu.

namespace ExampleNamespace;
public class MyClass
{
    public string MyProperty { get; set; }
}

reference:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces

**Comment by albert-github** [comment]
A relevant part from the mentioned link:

A file_scoped_namespace_declaration permits a namespace declaration to be written without the { ... } block. For example:
C#
```
extern alias A;
namespace Name;
using B;
class C
{
}
```
is semantically equivalent to
C#
```
extern alias A;
namespace Name
{
    using B;
    class C
    {
    }
}
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9333

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by romanchick8122** [comment]
Previous fix did not take into account nested namespaces, e.g.:
```
namespace A.B;
class C
{

}
```
was documented as so C is in global namespace.
I am proposing a fix to this, pull request #9528

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by romanchick8122** [comment]
As of version 1.9.5, the bug is fixed only partially. 
[Previously referenced article](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces) explicitly shows in its summary section that dots can be part of file-scoped namespace name. At the same time doxygen recognizes file-scoped namespace only if its name does not contain dots, and treats file contents as part of global namespace otherwise. [Attached example](https://github.com/doxygen/doxygen/files/9444496/example.tar.gz) has files showing doxygen behaviour in both cases.
For now, I am using my fix from pr #9528, but it would be nice to have this bug resolved in official release.

**Comment by KL-Sven** [comment]
I can confirm romanchick8122's statement. Please reopen this issue.

**Comment by albert-github** [comment]
@doxygen
Might also be related to https://github.com/doxygen/doxygen/issues/9445#issuecomment-1252293379 (and the patch #9528 might be better than the test I proposed with https://github.com/doxygen/doxygen/issues/9445#issuecomment-1252293379, I have not tested this though)

**Comment by albert-github** [comment]
Code of #9528 has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
